fe1-web/src/features/rollCall/network/__tests__/RollCallHandler.test.ts

```diff
@@ -0,0 +1,120 @@
+import 'jest-extended';
+import '__tests__/utils/matchers';
+import { getStore } from 'core/redux';
+import { getEventFromId } from 'features/events/network/EventHandlerUtils';
+import { Base64UrlData, Hash, Signature, Timestamp } from 'core/objects';
+import { addEvent, updateEvent } from 'features/events/reducer';
+import { EventTypeRollCall, RollCall, RollCallStatus } from 'features/rollCall/objects';
+import { ActionType, ObjectType } from 'core/network/jsonrpc/messages';
+import {
+  handleRollCallCloseMessage,
+  handleRollCallCreateMessage,
+  handleRollCallOpenMessage,
+  handleRollCallReopenMessage,
+} from '../RollCallHandler';
+
+jest.mock('core/network/JsonRpcApi');
+jest.mock('features/events/reducer');
+jest.mock('core/redux');
+jest.mock('features/events/network/EventHandlerUtils');
+
+const ID = new Hash('rollCallId');
+const NAME = 'myRollCall';
+const LOCATION = 'location';
+const TIMESTAMP_START = new Timestamp(1620255600);
+const TIMESTAMP_END = new Timestamp(1620357600);
+const ATTENDEES = ['attendee1', 'attendee2'];
+const rollCallStateCreated: any = {
+  id: ID.valueOf(),
+  eventType: EventTypeRollCall,
+  start: TIMESTAMP_START.valueOf(),
+  name: NAME,
+  location: LOCATION,
+  creation: TIMESTAMP_START.valueOf(),
+  proposedStart: TIMESTAMP_START.valueOf(),
+  proposedEnd: TIMESTAMP_END.valueOf(),
+  status: RollCallStatus.CREATED,
+};
+const rollCallStateOpened = {
+  ...rollCallStateCreated,
+  idAlias: ID.valueOf(),
+  openedAt: TIMESTAMP_START.valueOf(),
+  status: RollCallStatus.OPENED,
+};
+const rollCallStateClosed = {
+  ...rollCallStateOpened,
+  closedAt: TIMESTAMP_END.valueOf(),
+  status: RollCallStatus.CLOSED,
+  attendees: ATTENDEES,
+};
+const rollCallStateReopened = {
+  ...rollCallStateClosed,
+  status: RollCallStatus.REOPENED,
+};
+
+const mockRollCallClosed = RollCall.fromState(rollCallStateClosed);
+const mockRollCallCreated = RollCall.fromState(rollCallStateCreated);
+const mockRollCallOpened = RollCall.fromState(rollCallStateOpened);
+const mockRollCallReopened = RollCall.fromState(rollCallStateReopened);
+
+const mockMsg = (type: ActionType, status: RollCallStatus, rollCallState: any) => {
+  return {
+    laoId: ID,
+    receivedAt: TIMESTAMP_START,
+    channel: 'undefined',
+    data: Base64UrlData.encode(''),
+    sender: ID,
+    signature: new Signature(''),
+    message_id: ID,
+    witness_signatures: [],
+    messageData: {
+      object: ObjectType.ROLL_CALL,
+      action: type,
+      ...rollCallState,
+      proposed_start: rollCallState.proposedStart,
+      proposed_end: rollCallState.proposedEnd,
+      opened_at: rollCallState?.openedAt,
+      update_id: rollCallState?.idAlias,
+      closed_at: rollCallState?.closedAt,
+    },
+  };
+};
```

Comment:
I'd put all of this in a `region`. I allows us to collapse it to make it cleaner and to access the tests without having to scroll a lot. Basically, you just have to write `// region regionName` at the beginning of the data initialization, end an `// endregion` at the end. I'll let you name it however you want.